SYCL/ESIMD/esimd_test_utils.hpp

```diff
@@ -26,26 +26,26 @@ using namespace sycl;
 
 namespace esimd_test {
 
-// This is function provided to SYCL runtime by the application to decide
+// This is the function provided to SYCL runtime by the application to decide
 // on which device to run, or whether to run at all.
 // When selecting a device, SYCL runtime first takes (1) a selector provided by
 // the program or a default one and (2) the set of all available devices. Then
-// it passes each device to the selector. A device for which the highest number
-// is returned is selected. If a negative number was returned for all devices,
-// then the selection process will cause an exception.
+// it passes each device to the '()' operator of the selector. Device, for
+// which '()' returned the highest number, is selected. If a negative number
+// was returned for all devices, then the selection process will cause an
+// exception.
+// Require GPU device
 inline int ESIMDSelector(const device &device) {
-  if (const char *dev_filter = getenv("SYCL_DEVICE_FILTER")) {
-    std::string filter_string(dev_filter);
-    if (filter_string.find("gpu") != std::string::npos)
-      return device.is_gpu() ? 1000 : -1;
-
-    std::cerr
-        << "Supported 'SYCL_DEVICE_FILTER' env var device type is 'gpu' and "
-        << filter_string << "' does not contain that.\n";
-    return -1;
+  if (device.get_backend() == backend::ext_intel_esimd_emulator) {
```

Comment:
ESIMD tests can run only on either esimd emulator or Intel GPU device, all other devices should give negative score. So `return 0` below should be `return -1`, plus in addition to `device.is_gpu()` there should be Intel vendor check. `(device.is_gpu() && isVendorIntel(device))`